src/DependencyInjection/Configuration.php

```diff
@@ -40,8 +47,12 @@ private function configureRedis(ArrayNodeDefinition $node)
         ;
 
         $node->validate()->ifTrue(function ($nodeConfig) {
-            return isset($nodeConfig['host']) && isset($nodeConfig['service_id']);
-        })->thenInvalid('Only one of host and service_id must be provided');
+            return isset($nodeConfig['host'])
+                && isset($nodeConfig['service_id'])
+                && isset($nodeConfig['parameters'])
+                && isset($nodeConfig['options'])
+            ;
```

Comment:
Neither `options`, `parameters` nor `host` can be provided together with `service_id` - currently we check only if all three were provided with `service_id` configuration parameter.

Also we can have `options` with both `host` and `parameters`, but I think it's better if we check that only one of `parameters` and `host` is provided - this avoids unused configuration without any notification for developer.